### ISSUE TITLE
install: trusted dependency lifecycle script not executed

### ISSUE THREAD

**Issue by liz3** [oven-sh/bun#6580]
### What version of Bun is running?

1.0.6

### What platform is your computer?

darwin/linux

### What steps can reproduce the bug?

```sh
> mkdir some-folder
> cd some-folder
> bun init
# add bun-ui to trustedDependencies
> bun add https://github.com/liz3/bun-ui.git
``` 

### What is the expected behavior?

It should run the install script building bun-ui.

### What do you see instead?

It installs the package correctly but does not run the install script

### Additional information

This can be fixed by adding
```zig
manager.lockfile.trusted_dependencies = try lockfile.trusted_dependencies.clone(bun.default_allocator);
```
below https://github.com/oven-sh/bun/blob/main/src/install/install.zig#L8190

**Comment by liz3** [comment]
@Jarred-Sumner Maybe you can point out again where this should be copied if its not at the location i pointed out,

**Comment by dylan-conway** [comment]
This is fixed as of bun `v1.0.31`
